### PR TITLE
tests: Added tests to execute query

### LINKED ISSUE #12: Add tests for query
Assuming you have the following nodes:
```
{
	"nodes": [
		{
			"name": "apple",
			"labels": [
				"everyone",
				"apple",
				"slowdisk",
				"region=us-west-2",
			]
		},
		{
			"name": "banana",
			"labels": [
				"everyone",
				"banana",
				"region=us-west-2",
			]
		},
		{
			"name": "cherry",
			"labels": [
				"everyone",
				"cherry"
				"region=us-east-1",
			]
		}
	]
}
```

Then executing the queries should return:
- `apple`
  - `["apple"]`
- `(not ‘apple’)`
  - `["banana","cherry"]`
- `(and ‘slowdisk’ ‘region=us-west-2’)`
  -` ["apple"]`
- `(or ‘region=us-west-2’ ‘region=us-east-1’)`
  - `["apple","banana","cherry"]`
- `(or (not ‘slowdisk’) ‘banana’)`
  - `["banana","cherry"]`

### PR DESCRIPTION
**Pull request by sekerin**
Closes #12

### CODE REVIEW

File: query/execute_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+func TestExecute(t *testing.T) {
+	ctx := context.TODO()
+	ls := GetLabeled()
+
+	labeledSet, error := Execute(ctx, ls, "apple")
+	if error != nil {
+		return
+	}
+	require.Equal(t, []p2plab.Labeled{ls[0]}, labeledSet.Slice(), "Query apple return only apple node")
```
Comment (hinshun) [inline query/execute_test.go]: We should write these tests in a table format: https://github.com/golang/go/wiki/TableDrivenTests

File: query/execute_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		NewLabeled("banana", []string{"everyone", "banana", "region=us-west-2"}),
+		NewLabeled("cherry", []string{"everyone", "cherry", "region=us-east-1"}),
+	}
+}
+
+func TestExecute(t *testing.T) {
+	ctx := context.TODO()
+	ls := GetLabeled()
+
+	labeledSet, error := Execute(ctx, ls, "apple")
```
Comment (hinshun) [inline query/execute_test.go]: In golang, we usually use `err` as the variable name for errors. `error` is a reserved syntax for the error type, so I'm surprised you were able to use it as a variable name.

File: query/execute_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+func GetLabeled() []p2plab.Labeled {
+	return []p2plab.Labeled{
+		NewLabeled("apple", []string{"everyone", "apple", "slowdisk", "region=us-west-2"}),
+		NewLabeled("banana", []string{"everyone", "banana", "region=us-west-2"}),
+		NewLabeled("cherry", []string{"everyone", "cherry", "region=us-east-1"}),
+	}
+}
+
+func TestExecute(t *testing.T) {
+	ctx := context.TODO()
```
Comment (hinshun) [inline query/execute_test.go]: `context.Background()` is good enough here.

File: query/execute_test.go (hunk 1 above)
Comment (sekerin) [reply query/execute_test.go]: Big thanks, rewrited test with this pattern

File: query/execute_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func TestExecute(t *testing.T) {
+	ctx := context.Background()
+
+	for _, execute := range executetest {
+		labeledSet, err := Execute(ctx, ls, execute.in)
+		if err != nil {
+			return
+		}
```
Comment (hinshun) [inline query/execute_test.go]: We should check the error:
```go
labeledSet, err := Execute(ctx, ls, execute.in)
require.NoError(t, err)
```

**Comment by hinshun** [comment]
Thanks!

**Pull request merged**